.cloud/azuredeployBatchEndpoint.json

```diff
@@ -0,0 +1,228 @@
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "supportsHttpsTrafficOnly": true
+            }
+        },
+        {
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2019-09-01",
+            "name": "[parameters('keyVaultName')]",
+            "location": "[parameters('location')]",
+            "properties": {
+                "tenantId": "[subscription().tenantId]",
+                "sku": {
+                    "name": "standard",
+                    "family": "A"
+                },
+                "accessPolicies": []
+            }
+        },
+        {
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2018-05-01-preview",
+            "name": "[parameters('applicationInsightsName')]",
+            "location": "[if(or(equals(parameters('location'),'eastus2'), equals(parameters('location'),'centraluseuap')),'southcentralus',parameters('location'))]",
+            "kind": "web",
+            "properties": {
+                "Application_Type": "web"
+            }
+        },
+        {
+            "type": "Microsoft.MachineLearningServices/workspaces",
+            "apiVersion": "2020-03-01",
+            "name": "[parameters('workspaceName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[parameters('applicationInsightsName')]",
+                "[parameters('keyVaultName')]",
+                "[parameters('storageAccountName')]"
+            ],
+            "sku": {
+                "tier": "enterprise",
+                "name": "enterprise"
+            },
+            "identity": {
+                "type": "systemAssigned"
+            },
+            "properties": {
+                "friendlyName": "[parameters('workspaceName')]",
+                "storageAccount": "[variables('storageAccountResourceId')]",
+                "keyVault": "[variables('keyVaultResourceId')]",
+                "applicationInsights": "[variables('appInsightsResourceId')]"
+            }
+        },
+        {
+            "type": "Microsoft.MachineLearningServices/workspaces/batchEndpoints",
+            "apiVersion": "2020-09-01-preview",
+            "location": "[parameters('location')]",
+            "name": "[concat(parameters('workspaceName'), '/', parameters('batchEndpointName'))]",
+            "dependsOn": [
+                "[parameters('workspaceName')]"
+            ],
+            "identity": {
+                "type": "None"
+            },
+            "properties": {
+                "name": "[concat(parameters('workspaceName'), '/', parameters('batchEndpointName'))]",
+                "description": "Testing batch endpoint creation"
+            }
+        },
+        {
+            "type": "Microsoft.MachineLearningServices/workspaces/batchEndpoints/deployments",
```

Comment:
we are missing 
1) compute
2) model
3) data
3) env
here

Reply:
do you mean creating these entities here? 
- Compute, yes, we can do it. let me look at what code job does here.
- The model will be a trained one, not sure what you would even provide here, given that the training will be done separately.
- Data is not relevant for a batch endpoint creation.
- Env in the MLflow model case, is an optional param, hence not specified here.